pkg/pillar/cmd/domainmgr/containerd.go

```diff
@@ -197,3 +288,61 @@ func getRktPodManifest(PodManifestFile string) (RktPodManifest, error) {
 	}
 	return manifest, nil
 }
+
+func loadContainer(containerID string) (containerd.Container, error) {
+	ctrdClient, err := getContainerdClient(ctrdSocket)
+	if err != nil {
+		log.Errorf("Could not build containerd client. %v", err.Error())
+		return nil, fmt.Errorf("Exception while loading container. %v", err)
+	}
+	defer ctrdClient.Close()
+	ctrdCtx := getContainerdContext(CtrdServicesNamespace)
+	container, err := ctrdClient.LoadContainer(ctrdCtx, containerID)
+	if err != nil {
+		return nil, fmt.Errorf("Exception while loading container. %v", err)
+	}
+	return container, nil
+}
+
+func getContainerdClient(ctrdSocketPath string) (*containerd.Client, error) {
+	ctrdClient, err := containerd.New(ctrdSocketPath)
+	if err != nil {
+		log.Errorf("could not connect to containerd socket at %s: %+s", ctrdSocketPath, err.Error())
+		return nil, fmt.Errorf("could not connect to containerd socket at %s: %+s", ctrdSocketPath, err.Error())
+	}
+	return ctrdClient, nil
+}
+
+func getContainerdContext(ctrdNameSpace string) context.Context {
+	ctrdCtx := namespaces.WithNamespace(context.Background(), ctrdNameSpace)
+	return ctrdCtx
+}
+
+func getImage(imageName string) (containerd.Image, error) {
+	var err error
+	ctrdClient, err := getContainerdClient(ctrdSocket)
+	if err != nil {
+		log.Errorf("Could not build containerd client. %v", err.Error())
+		return nil, fmt.Errorf("Exception while loading container. %v", err)
+	}
+	defer ctrdClient.Close()
+	ctrdCtx := getContainerdContext(CtrdServicesNamespace)
+	return ctrdClient.GetImage(ctrdCtx, imageName)
+}
+
+// Util methods
+
+func buildEnvVarsSlice(envMap map[string]string) []string {
+	var envVars []string
+
+	for envName, envValue := range envMap {
+		envVars = append(envVars, fmt.Sprintf("%s=%s", envName, envValue))
+	}
+
+	return envVars
+}
+
+func buildContainerID(displayName string) string {
+	uuid := uuid.NewV4().String()
+	return fmt.Sprintf("%s-%s", displayName, uuid)
```

Comment:
This is good as a function. Do we need to include the displayName in the container name? I would suggest we include just the generated UUID, and if we need anything else, include it as a label on the container, see [this option](https://godoc.org/github.com/containerd/containerd#WithContainerLabels)